ComplexProperties/TimeZones/TimeZoneTransitionGroup.cs

```diff
@@ -130,7 +130,7 @@ internal virtual void InitializeFromAdjustmentRule(AdjustmentRule adjustmentRule
                     adjustmentRule.DateStart.Year);
                 standardPeriodToSet.Name = standardPeriod.Name;
                 standardPeriodToSet.Bias = standardPeriod.Bias;
-                this.timeZoneDefinition.Periods.Add(standardPeriodToSet.Id, standardPeriodToSet);
+                this.timeZoneDefinition.Periods.AddOrUpdate(standardPeriodToSet.Id, standardPeriodToSet);

```

Comment:
this.timeZoneDefinition.Periods.
```suggestion
                this.timeZoneDefinition.Periods[standardPeriodToSet.Id] = standardPeriodToSet;
```